### ISSUE TITLE
Bad DataType.defaultValue() call generated for MySQL tables

### ISSUE THREAD

**Issue by karolhor** [jOOQ/jOOQ#5291]
Hi,
I've generated the schema from MySQL. I'm trying to DSLContext.ddl() with HSQLDB in my integration tests, but varchar column with default value in generated SQL for CREATE TABLE is not quoted.

```java
Caused by: org.jooq.exception.DataAccessException: SQL [create table "users"(
  "ID" bigint not null,
  "FIRST_NAME" varchar(36) not null,
  "LAST_NAME" varchar(36) not null,
  "COUNTRY" varchar(30) not null default GERMANY,
  constraint "KEY_users_PRIMARY"
    primary key ("ID")
)]; unexpected token: GERMANY : line: 5
at org.jooq.impl.Tools.translate(Tools.java:1907)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:659)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.impl.DefaultDSLContext.execute(DefaultDSLContext.java:3257)
...
```

In Users table class I have:
```java
public final TableField<UsersRecord, String> COUNTRY = 
    createField("COUNTRY", org.jooq.impl.SQLDataType.VARCHAR.length(30).nullable(false).defaultValue(org.jooq.impl.DSL.field("GERMANY", org.jooq.impl.SQLDataType.VARCHAR)), this, "");
```

According to http://www.hsqldb.org/doc/guide/ch09.html#create_table-section 
default value for character column should be **single** quoted.

**Comment by lukaseder** [comment]
Interesting. I suspect that your Users table which contains that field definition is generated by the code generator?

That would be a bug then. The literal shouldn't be wrapped with `DSL.field()`, but with `DSL.inline()` (as a workaround, you could manually fix this)

**Comment by karolhor** [comment]
yes, my code was generated by the code generator. 

for now I added single quote in the field definition and it works. 
```java
org.jooq.impl.DSL.field("'GERMANY'", org.jooq.impl.SQLDataType.VARCHAR)
```

**Comment by lukaseder** [comment]
Looks like this issue is database-specific. Some databases already provide valid expressions for `DSL.field()` (e.g. PostgreSQL). Others provide the actual value (e.g. MySQL)

**Comment by lukaseder** [comment]
OK, fixed this for MySQL only. Most other databases seem to report executable expressions, not the actual values. This makes sense, because column defaults are really expressions, such as `CURRENT_TIMESTAMP()`.

The fix will be backported to jOOQ 3.8.2. Thanks again for reporting!
